### PR TITLE
SPIR-V image operations

### PR DESCRIPTION
**Pull request by expipiplus1**

### CODE REVIEW

File: source/slang/slang-stdlib-textures.cpp [hunk 1]
```diff
@@ -0,0 +1,1213 @@
...
+            sb << spirv << "\n";
+        }
+    }
+}
+
+void TextureTypeInfo::writeFuncDecorations(
+    const String& glsl,
+    const String& cuda)
+{
+    if(glsl.getLength())
```
Comment (csyonghe) [inline source/slang/slang-stdlib-textures.cpp:99]: We no longer need to write target_intrinsics, right?

File: source/slang/slang-stdlib-textures.cpp [hunk 2]
```diff
@@ -0,0 +1,1213 @@
...
+    spirvBuilder << " None";
+    if(bias)
+        spirvBuilder << "|Bias";
+    if(lodLevel)
+        spirvBuilder << "|Lod";
+    if(gradX)
+        spirvBuilder << "|Grad";
+    if(constOffset)
+        spirvBuilder << "|ConstOffset";
+    if(minLod)
```
Comment (csyonghe) [inline source/slang/slang-stdlib-textures.cpp:801]: Also need `Sample` before `minlod`.

File: source/slang/slang-stdlib-textures.cpp (hunk 1 above)
Comment (expipiplus1) [reply source/slang/slang-stdlib-textures.cpp:99]: Yeah, I had it this way to minimize a massive diff from the beginning while working on this, just to verify that the refactoring was correct. I also wonder if it's slightly faster to process the decorations version, but haven't confirmed.

File: source/slang/slang-stdlib-textures.cpp (hunk 2 above)
Comment (expipiplus1) [reply source/slang/slang-stdlib-textures.cpp:801]: Yeah, I think this bit will get cleaned up, it was just implemented lazily so far.

**Pull request merged**